src/SocialAPIDataHandler.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Drupal\social_api;
+
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+
+/**
+ * Variables are written to and read from session via this class.
+ */
+class SocialAPIDataHandler {
```

Comment:
Can we make it abstract, so only SocialAuthDataHandler and SocialPostDataHandler can be instantiated (assuming both extend this class)?